<a href="https://colab.research.google.com/github/dev-gustavom/Aurora_Singer/blob/main/Aurora_Singer.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [9]:
from google.colab import files

# Upload do arquivo CSV
uploaded = files.upload()

Saving telemetria.csv to telemetria (1).csv


In [10]:
import pandas as pd

# Ler o CSV
df = pd.read_csv("telemetria.csv")

# Mostrar os dados
df

,tempo_s,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar,status_modulos
0,0,22,-10,1,95,250,OK
1,10,23,-12,1,93,248,OK
2,20,24,-15,1,90,245,OK
3,30,25,-18,1,87,240,OK
4,40,26,-20,1,85,238,OK
5,50,27,-25,1,82,235,OK
6,60,28,-30,1,80,230,OK


In [11]:
filename = list (uploaded.keys()) [0]
df = pd.read_csv(filename, sep= ';', decimal= ',')
df.head()

,"tempo_s,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar,status_modulos"
0,"0,22,-10,1,95,250,OK"
1,"10,23,-12,1,93,248,OK"
2,"20,24,-15,1,90,245,OK"
3,"30,25,-18,1,87,240,OK"
4,"40,26,-20,1,85,238,OK"


In [12]:
# Última linha da tabela
ultima_linha = df.iloc[-1]

print(ultima_linha)

tempo_s,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar,status_modulos    60,28,-30,1,80,230,OK
Name: 6, dtype: object


In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7 entries, 0 to 6
Data columns (total 1 columns):
 #   Column                                                                                                                          Non-Null Count  Dtype 
---  ------                                                                                                                          --------------  ----- 
 0   tempo_s,temperatura_interna_c,temperatura_externa_c,integridade_estrutural,nivel_energia_pct,pressao_tanque_bar,status_modulos  7 non-null      object
dtypes: object(1)
memory usage: 188.0+ bytes


# **Descrição dos dados:**
- temperatura_interna_c → temperatura dentro da nave
- temperatura_externa_c → ambiente externo
- integridade_estrutural → 1 = OK | 0 = falha
- nivel_energia_pct → bateria (%)
- pressao_tanque_bar → pressão do combustível
- status_modulos → OK ou FALHA

# **Pseudocódigo – Algoritmo de Verificação**

INICIO

LER dados da telemetria


SE temperatura_interna < 18 OU temperatura_interna > 30

    RETORNAR "DECOLAGEM ABORTADA"


SE integridade_estrutural == 0

    RETORNAR "DECOLAGEM ABORTADA"


SE nivel_energia < 80

    RETORNAR "DECOLAGEM ABORTADA"


SE pressao_tanque < 230 OU pressao_tanque > 260

    RETORNAR "DECOLAGEM ABORTADA"


SE status_modulos != "OK"

    RETORNAR "DECOLAGEM ABORTADA"


CASO CONTRÁRIO

    RETORNAR "PRONTO PARA DECOLAR"


FIM

# **Script em Python**

# Simulação de leitura de dados (última linha da tabela)
temperatura_interna = 28

temperatura_externa = -30

integridade_estrutural = 1

nivel_energia = 80

pressao_tanque = 230

status_modulos = "OK"

# Verificação das condições

# Verifica temperatura interna segura
if temperatura_interna < 18 or temperatura_interna > 30:

    print("DECOLAGEM ABORTADA - Temperatura interna fora do limite")

# Verifica integridade estrutural
elif integridade_estrutural == 0:

    print("DECOLAGEM ABORTADA - Falha estrutural")

# Verifica nível de energia
elif nivel_energia < 80:

    print("DECOLAGEM ABORTADA - Energia insuficiente")

# Verifica pressão do tanque
elif pressao_tanque < 230 or pressao_tanque > 260:

    print("DECOLAGEM ABORTADA - Pressão inadequada")

# Verifica status dos módulos
elif status_modulos != "OK":

    print("DECOLAGEM ABORTADA - Problema nos módulos")

# Se tudo estiver correto
else:

    print("PRONTO PARA DECOLAR")

# **Análise Energética**

Considerando:

Capacidade total: 1000 kWh

Carga atual: 80%

Consumo na decolagem: 300 kWh

Perdas energéticas: 10%

In [14]:
# Dados iniciais
capacidade_total = 1000  # kWh (capacidade máxima da bateria)
carga_atual_pct = 80     # porcentagem atual
consumo_decolagem = 300  # energia usada na decolagem (kWh)
perda_pct = 10           # perdas energéticas (%)

# Converter porcentagem em valor real
energia_disponivel = capacidade_total * (carga_atual_pct / 100)

# Calcular perdas
perdas = energia_disponivel * (perda_pct / 100)

# Energia útil após perdas
energia_util = energia_disponivel - perdas

# Energia restante após decolagem
energia_restante = energia_util - consumo_decolagem

# Mostrar resultados
print("Energia disponível:", energia_disponivel, "kWh")
print("Perdas energéticas:", perdas, "kWh")
print("Energia útil:", energia_util, "kWh")
print("Energia restante após decolagem:", energia_restante, "kWh")

# Verificar se há energia suficiente para decolagem

if energia_restante < 0:
    print("Energia insuficiente para decolagem!")
elif energia_restante < 200:
    print("Decolagem possível, mas com baixo nível de segurança!")
else:
    print("Energia suficiente para decolagem segura!")

Energia disponível: 800.0 kWh
Perdas energéticas: 80.0 kWh
Energia útil: 720.0 kWh
Energia restante após decolagem: 420.0 kWh
Energia suficiente para decolagem segura!


# **Análise Assistida por IA**
# Classificação dos dados

# Com base na tabela de telemetria:

Temperatura interna (22–28°C) → Dentro da faixa segura (NORMAL)

Temperatura externa (-10 a -30°C) → Condição extrema, mas esperada (ACEITÁVEL)

Integridade estrutural (1) → Sem falhas (CRÍTICO OK)

Nível de energia (95% → 80%) → Tendência de queda, mas ainda suficiente (ATENÇÃO)

Pressão do tanque (250 → 230 bar) → Dentro do limite mínimo (LIMITE INFERIOR)

Status dos módulos (OK) → Sistemas operacionais (NORMAL)

Identificação de possíveis anomalias

Queda contínua do nível de energia

→ Pode indicar consumo elevado ou ineficiência energética
Redução gradual da pressão do tanque

→ Possível vazamento ou consumo acima do previsto
Temperatura externa muito baixa

→ Pode afetar componentes sensíveis e eficiência dos sistemas

# Sugestões de risco

**Risco energético**

→ Monitorar consumo durante a decolagem para evitar falha crítica

**Risco de pressão**

→ Verificar válvulas e sistema de combustível antes da decolagem

**Risco térmico**

→ Garantir isolamento térmico e aquecimento interno adequado

**Risco operacional**

→ Implementar alertas automáticos para mudanças rápidas nos dados

# **Reflexão Crítica**

A exploração espacial levanta importantes questões éticas e sociais. O investimento em missões espaciais deve ser equilibrado com as necessidades da população na Terra, como saúde, educação e combate à pobreza.

Ao mesmo tempo, a exploração do espaço impulsiona o desenvolvimento tecnológico, gerando avanços que beneficiam toda a sociedade. Tecnologias como GPS, painéis solares e sistemas de comunicação são exemplos disso.

A responsabilidade também envolve evitar a contaminação de outros planetas e preservar o ambiente espacial. A sustentabilidade tecnológica deve ser uma prioridade, garantindo que os recursos sejam utilizados de forma consciente e eficiente.

Além disso, é fundamental que os benefícios da exploração espacial sejam acessíveis globalmente, e não apenas concentrados em poucos países ou empresas.

Em resumo, explorar o espaço é importante, mas deve ser feito com responsabilidade, ética e foco no bem coletivo.